src/test/java/games/strategy/debug/LogReaderTest.java

```diff
@@ -28,7 +29,7 @@
   public void testStreamSplittingArray() throws Exception {
     final LogReader reader = new LogReader(stream, area, Boolean.TRUE::booleanValue, console);
     final String testString = "Some Test String";
-    final byte[] testByteArray = testString.getBytes();
+    final byte[] testByteArray = testString.getBytes(Charset.defaultCharset());
```

Comment:
I used the default charset in this test fixture because the target stream is always wrapped in a `PrintStream` (in `GenericConsole`), which hard-codes usage of the default charset.

Reply:
Why default and not UTF-8? Would that lower risk of OS specific charset problems?

Reply:
`PrintStream` always encodes a string using the default charset (presumably because its sink is typically stdout or stderr).  We are writing to the same sink, so we should be consistent with the behavior of the outer `PrintStream` so we don't (potentially) use two different encodings in the same stream.

Reply:
Here's an example of what can go wrong if the default charset is not used to write to stdout/stderr:

```java
import java.io.*;
import java.nio.charset.*;

public final class PrintStreamTest {
    public static void main(final String[] args) throws Exception {
      final String textWithNonAsciiChars = "ABCáéíóú123";
      final PrintStream ps1 = new PrintStream(System.out, true, Charset.defaultCharset().name());
      ps1.println(textWithNonAsciiChars);
      final PrintStream ps2 = new PrintStream(System.out, true, StandardCharsets.ISO_8859_1.name());
      ps2.println(textWithNonAsciiChars);
    }
}
```

The default charset on my system is UTF-8.  Running the above program in a console window produces:

```
$ java PrintStreamTest
ABCáéíóú123
ABC�����123
```

Piping the output through `iconv` shows the result if I were to change my console's charset to ISO-8859-1:

```
$ java PrintStreamTest | iconv -f iso-8859-1
ABCÃ¡Ã©Ã­Ã³Ãº123
ABCáéíóú123
```

The bottom line is that we have no control over how the user configures the encoding for stdout and stderr, and so using the default charset in `LogReader` is probably the right thing to do.  Alternatively, since the decision to pass stdout/stderr to `LogReader` is made by `GenericConsole`, we could require the charset to use to be passed to `LogReader` explicitly.  But since `LogReader` is only used by `GenericConsole`, that may be overkill at this time.

Reply:
Thank you for the context, with that I would agree with the decision to use default encoding in this case.